app/src/chainlit.py

```diff
@@ -127,11 +127,22 @@ async def on_message(message: cl.Message) -> None:
     engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
     try:
         result = await cl.make_async(lambda: engine.on_message(question=message.content))()
-        msg_content = result.response + format_guru_cards(
-            docs_shown_max_num=engine.docs_shown_max_num,
-            docs_shown_min_score=engine.docs_shown_min_score,
-            chunks_with_scores=result.chunks_with_scores,
-        )
+
+        msg_content = result.response
+
+        if isinstance(engine, BridgesEligibilityManualEngine):
```

Comment:
Consider: Instead of importing a specific chat engine (`BridgesEligibilityManualEngine`), could we call `engine.formatting` that returns a string or callable, which would determine which UI formatting should be used for the engine? If it returns a string, then that string could be passed to a proxy `format_retrieved_docs()` function that would call the desired formatting function (`format_bem_documents()` or `format_guru_cards()`).